traffic_ops/traffic_ops_golang/deliveryservice/acme.go

```diff
@@ -306,17 +355,26 @@ func GetAcmeCertificates(cfg *config.Config, req tc.DeliveryServiceLetsEncryptSS
 	if err != nil {
 		log.Errorf("deliveryservice.GenerateSSLKeys: getting DS ID from name " + err.Error())
 		api.CreateChangeLogRawTx(api.ApiChange, "DS: "+*req.DeliveryService+", ID: "+strconv.Itoa(dsID)+", ACTION: FAILED to add SSL keys with "+provider, currentUser, logTx)
+		if err = api.UpdateAsyncStatus(db, api.AsyncFailed, "ACME renewal failed.", asyncStatusId, true); err != nil {
+			log.Errorf("updating async status for id %v: %v", asyncStatusId, err)
+		}
 		return errors.New("deliveryservice.GenerateSSLKeys: getting DS ID from name " + err.Error())
```

Comment:
This situation is one of the reasons I tend to prefer `fmt.Errorf` to `errors.New`.

If getting the Delivery Service ID from its XMLID fails with some error, this will log the error, enter a changelog entry, update the async job status, and then segfault - _unless_ updating the async job status also fails. Because you're re-assigning `err` based on the output of `api.UpdateAsyncStatus`, if that succeeds `err` will be `nil` and `err.Error` is an invalid dereference